src/lib/network_pool/transaction_pool.ml

```diff
@@ -1392,16 +1395,45 @@ struct
           register_event
             { msg = "Received transaction-pool diff $txns from $sender" }]
 
-      let update_metrics envelope valid_cb gossip_heard_logger_option =
+      let update_metrics ~logger ~log_gossip_heard envelope valid_cb =
         Mina_metrics.(Counter.inc_one Network.gossip_messages_received) ;
         Mina_metrics.(Gauge.inc_one Network.transaction_pool_diff_received) ;
         let diff = Envelope.Incoming.data envelope in
-        Option.iter gossip_heard_logger_option ~f:(fun logger ->
-            [%str_log debug]
-              (Transactions_received
-                 { txns = diff; sender = Envelope.Incoming.sender envelope } ) ) ;
+        if log_gossip_heard then
+          [%str_log debug]
+            (Transactions_received
+               { txns = diff; sender = Envelope.Incoming.sender envelope } ) ;
         Mina_net2.Validation_callback.set_message_type valid_cb `Transaction ;
         Mina_metrics.(Counter.inc_one Network.Transaction.received)
+
+      let log_internal ?reason ~logger msg
+          { Envelope.Incoming.data = diff; sender; _ } =
+        let metadata =
+          [ ( "diff"
+            , `List
+                (List.map diff
+                   ~f:Mina_transaction.Transaction.yojson_summary_of_command )
+            )
+          ]
+        in
+        let metadata =
+          match sender with
+          | Remote addr ->
+              ("sender", `String (Core.Unix.Inet_addr.to_string @@ Peer.ip addr))
+              :: metadata
+          | Local ->
+              metadata
```

Comment:
I thought about this. My idea was that the format of this internal log is well-defined, for this reason we can be sure that absence of the field is no field at all. I was hesitant to use `local` because as of now if there is a `sender`, we know it's a host name. If we use `local`, then we would add one more case to interpreting the string. I didn't find it to be too useful. These logs are machine-readable after all and are not intended to be read manually.